private/buf/bufgen/config.go

```diff
@@ -43,44 +45,47 @@ func readConfig(
 	if override := readConfigOptions.override; override != "" {
 		switch filepath.Ext(override) {
 		case ".json":
-			return getConfigJSONFile(override)
+			return getConfigJSONFile(logger, override)
 		case ".yaml", ".yml":
-			return getConfigYAMLFile(override)
+			return getConfigYAMLFile(logger, override)
 		default:
-			return getConfigJSONOrYAMLData(override)
+			return getConfigJSONOrYAMLData(logger, override)
 		}
```

Comment:
Letting `getConfigJSONFile`, `getConfigYAMLFile` and `getConfigJSONOrYAMLData` use `provider.logger` would  also have worked (and would've reduced the number of changes), but I don't think that's good, because that logger is meant for `provider` only.

Reply:
Defining something like a `ConfigReader` and injecting the logger into it could potentially reduce the need to pass the logger down through 3 function calls, but I'm not sure if that's necessary.

Reply:
Yes, I think you are correct in not adding the logger to any of the config structures, since it is not extensively used. I agree with this approach 👍🏼